Gifski/EditVideoViewController.swift

```diff
@@ -324,6 +325,68 @@ final class EditVideoViewController: NSViewController {
 		}
 	}
 
+	private func showKeyframeRateWarningIfNeeded() {
+		DispatchQueue.global(qos: .utility).async { [weak self] in
+			guard
+				Defaults[.bounceGif],
+				let asset = self?.asset,
+				let videoTrack = asset.firstVideoTrack,
+				let reader = try? AVAssetReader(asset: asset)
+			else {
+				return
+			}
+
```

Comment:
I moved the logic to an extension on `AVAssetTrack`. I made the method synchronous since the logic itself is synchronous. The caller of the method can dispatch to a background queue if needed.